docs/api/v1.md

```diff
@@ -0,0 +1,383 @@
+    - Returns {[Error](#error)]}
+
+
+## Object Definitions
+
+
+### Error
+```
+{
+   code {integer}: Unique code identifying this error type
+   name {string}: Unique human readable name for this error type
+   message {string}: Human readable error message
+   data {object}: Data specific to this error. See [Error Codes](#error-codes) for specifics.
+}
+```
+
+
+
+
+### File
+```
+{
+    name {string}: Name of file
+    path {string}: Path to file
+    type {string}: File type ("file" or "directory")
+    contents: {object[]}:
+        If type="file", undefined
+        If type="directory", an array of Files inside the directory. Directories have "contents" field set to undefined.
+}
+```
+
+
+### Process
+```
+{
+    id {int}: Identifying number of process on host machine
+    name {string}: Name of process on host machine
+}
+```
+
+
+### Execution
+```
+{
+   id {integer}: Unique Identifier
+   type {string}: Type of execution. Either `function` or `process`
+   status {string}: Either `pending`, `executing`, `stopped`, or `done`
+   executionTime {integer}: Nanoseconds
+   data {object}: Data specific to this type
+      if `type=function`: {
+         function {[Function](#function)}: Function that produced this execution
+      }
+      if `type=process`: {
+         nextExecution {Execution id}: Id of the execution that follows this one. Null until `status=done`. Initially set to `null`, set when a call to `/debug/execute` hits a breakpoint, in which case it points to the breakpoint function�s execution.
+      }
+}
+```
+
+
+### Trace
+```
+{
+   index {integer}: Index of the trace, beginning at 0 and totally ordered for each execution
+   type {integer}: Type of trace: `0`=instruction (state changes), `1`=output (to stdout), `2`=termination (data about how the execution ended)
+   line {integer}: Line number that produced this trace
+   data {object}: Data specific to the trace type
+      if `type=0`: {
+         state {object[]}: Array of variable changes {variable {Variable}, value {*}}
+      }
+      If `type=1`: {
+         output {string}: Stdout output
+      }
+      If `type=2`: {
+         cause {string}: Reason for termination. Either `stopped`, `crashed`, `ended`, or `breakpoint`
+         stack {string} (defined if cause=crashed): Stack trace output
+         code {integer} (defined if cause=ended): Program exit code
+         nextExecution {Execution id} (defined if cause=breakpoint): Id of the following function execution
+      }
+}
+```
+
+
+### Function
+```
+{
+   address {integer}: Memory address of the function
+   name {string}: Name of the function
+   sourcePath {string}: Full path to the source code
+   lineNumber {integer}: Line number of the function definition
+   lineCount {integer}: Length of function in source code lines
+   parameters {object[]}: Array of function parameter objects--{name {string}, type {SourceType}}
+}
+```
+### AttachInfo
+```
+{
+  attachedProcess {[Process](#process)}
+}
```

Comment:
this is a nit pick but rust warns against using camel case and prefers snake_case.  I could be convinced to stick with camel but it would be cool to go with snake_case to get rid of the compiler warnings.